src/west/manifest.py

```diff
@@ -66,6 +66,9 @@
 # though it's just a str in each individual file right now.
 WestCommandsType = Union[str, List[str]]
 
+# Type for the submodules value passed through the manifest file.
+SubmodulesType = Union[str, List[str]]
```

Comment:
Shouldn't it be `SubmodulesType = List[Dict]`? Have you tried to pass the file through `mypy`? (you can install it with `pip3 install mypy` and then run `mypy manifest.py`)